### PR TITLE
chore: Jupyter tests

### PR DESCRIPTION
**Pull request by DanielSchiavini**
### What I did
Separate job to run the jupyter tests with proper dependencies installed.

### How I did it
- Add optional dependencies for testing jupyter
- Updated the tests

### How to verify it
Plugin should keep working. Unit tests should be green.

### Description for the changelog
N/A

### Cute Animal Picture
image

### CODE REVIEW

File: tests/unitary/jupyter/jupyter_server/base/handlers.py [hunk 1]
```diff
@@ -0,0 +1,2 @@
+class APIHandler:
```
Comment (DanielSchiavini) [inline tests/unitary/jupyter/jupyter_server/base/handlers.py:1]:
```suggestion
"""
Fake jupyter_server.base module for testing without Jupyter server installed.
"""
class APIHandler:
```

File: tests/integration/jupyter/conftest.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+from multiprocessing.shared_memory import SharedMemory
+from os import urandom
+
+import pytest
+
+
+@pytest.fixture()
+def shared_memory_length():
+    from boa.integrations.jupyter.constants import SHARED_MEMORY_LENGTH
```
Comment (charles-cooper) [inline tests/integration/jupyter/conftest.py:9]: hm, why do we need inline imports?

File: pyproject.toml [hunk 3]
```diff
@@ -34,6 +34,7 @@ dependencies = [
 
 [project.optional-dependencies]
 forking-recommended = ["ujson"]
+jupyter-testing = ["jupyter_server", "ipython", "nest_asyncio"]
```
Comment (charles-cooper) [inline pyproject.toml:37]: i'd rather we put this in a "test requirements" (or just throw them into `dev-requirements.txt`)

Comment (DanielSchiavini) [reply pyproject.toml:37]: Then we don't need a separate job as we agreed, right?

File: tests/unitary/jupyter/conftest.py [hunk 4]
```diff
@@ -1,74 +1,26 @@
...
-    mock = MagicMock()
-    mock.authenticated = lambda x: x
-    replace_modules({"nest_asyncio": mock})
-    return mock
-
-
-@pytest.fixture()
-def shared_memory_length(nest_asyncio_mock):
+def shared_memory_length():
     from boa.integrations.jupyter.constants import SHARED_MEMORY_LENGTH
```
Comment (charles-cooper) [inline tests/unitary/jupyter/conftest.py:10]: another inline import

Comment (DanielSchiavini) [reply tests/unitary/jupyter/conftest.py:10]: I don't know how I missed this one :scream:

File: tests/unitary/jupyter/conftest.py [hunk 5]
```diff
@@ -1,74 +1,26 @@
...
-def nest_asyncio_mock(replace_modules):
-    mock = MagicMock()
-    mock.authenticated = lambda x: x
-    replace_modules({"nest_asyncio": mock})
-    return mock
-
-
-@pytest.fixture()
-def shared_memory_length(nest_asyncio_mock):
+def shared_memory_length():
```
[the remark this answers is not in the document]
Comment (DanielSchiavini) [reply tests/unitary/jupyter/conftest.py:9]: nope, not anymore.

File: dev-requirements.txt [hunk 6]
```diff
@@ -12,3 +12,6 @@ pytest
 pytest-xdist
 pytest-cov
 sphinx-rtd-theme
+jupyter_server
+ipython
```
Comment (charles-cooper) [inline dev-requirements.txt:16]: dup

File: dev-requirements.txt [hunk 7]
```diff
@@ -12,3 +12,6 @@ pytest
...
+nest_asyncio
```
Comment (charles-cooper) [inline dev-requirements.txt:17]: nest_asyncio comes with jupyter, no?

File: dev-requirements.txt [hunk 8]
```diff
@@ -12,3 +12,6 @@ pytest
 pytest-xdist
 pytest-cov
 sphinx-rtd-theme
+jupyter_server
```
Comment (charles-cooper) [inline dev-requirements.txt:15]: maybe add a comment or make it a new section
```suggestion

# jupyter
jupyter_server
```

**Review by charles-cooper — APPROVED** [review]

**Pull request merged**